en/Search Algorithms/Floyd Cycle Detection Algorithm to Find Duplicate Number.md

```diff
@@ -0,0 +1,38 @@
+# Floyd Cycle Detection Algorithm to find duplicate number in an array
+
+## Problem Statement
+
+Given an array of integers containing `n + 1` integers, where each integer is in the range `[1, n]` inclusive. If there is only one duplicate number in the input array, this algorithm returns the duplicate number without modifying the original array, otherwise, it returns -1.
+
+## Approach
+
+- Use the function `f(x) = arr[x]` to construct the sequence: 
+`arr[0]`, `arr[arr[0]]`, `arr[arr[arr[0]]]`, `arr[arr[arr[arr[0]]]]`, etc....
+- Each new element in the sequence is an element in `arr[]` at the index of the previous element.
+- Starting from `x = arr[0]`, it will produce a linked list with a cycle.
```

Comment:
Yeah, but *how do you actually detect the cycle*?

Reply:
I think you have a good point here.

@Swastyy the cycle is detected when `tortoise == hare`.

Reply:
Yes, that's what we have implemented when tortoise will be equal to hare it implies that there is a cycle or a duplicate number.